components/dashboard/src/App.tsx

```diff
@@ -159,14 +159,18 @@ function App() {
     const { user, setUser, refreshUserBillingMode } = useContext(UserContext);
     const { teams, setTeams } = useContext(TeamsContext);
     const { setIsDark } = useContext(ThemeContext);
-    const { usePublicApiTeamsService } = useContext(FeatureFlagContext);
+    const { usePublicApiTeamsService, useSlowDatabase } = useContext(FeatureFlagContext);
 
     const [loading, setLoading] = useState<boolean>(true);
     const [isWhatsNewShown, setWhatsNewShown] = useState(false);
     const [showUserIdePreference, setShowUserIdePreference] = useState(false);
     const [isSetupRequired, setSetupRequired] = useState(false);
     const history = useHistory();
 
+    useEffect(() => {
+        initGitPodService(useSlowDatabase);
+    }, [useSlowDatabase]);
```

Comment:
This seems to cause a bootstrap problem. We don't know who the user is (to evaluate the feature flag), but we're creating the connection which is parametrized on the feature flag (by user). I suspect this would result in always using "false" for the slow database argument

Reply:
The value of `useSlowDatabase` is taken from the `FeatureFlagContext`. All feature flags are re-evaluated when the value of `user` changes (the dependency array in the effect includes `user`):

https://github.com/gitpod-io/gitpod/blob/46c67229f31c583822f9009d28603e1f1fc0bdc2/components/dashboard/src/contexts/FeatureFlagContext.tsx#L47-L85

Whenever the context sets the value of `useSlowDatabase` (as it will whenever the effect runs) it causes any component that consumes the context to re-render, and the dependency array in the `useEffect` here ensures that we re-initialize the websocket connection whenever `App` renders.

Reply:
I guess with this change, we might see nearly double the WS connections being created. Just something we should be aware of from ops side

Reply:
I think we'll only see two connections for the small subset of users for which the feature flag is enabled.

For everyone else, the value of `useSlowDatabase` will never change and the connection will never be re-initialized.